compiler/optimizer/abstractinterpreter/IDT.cpp

```diff
@@ -0,0 +1,182 @@
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "optimizer/abstractinterpreter/IDT.hpp"
+
+TR::IDT::IDT(TR::Region& region, TR_CallTarget* callTarget, TR::ResolvedMethodSymbol* symbol, uint32_t budget, TR::Compilation* comp):
+      _region(region),
+      _maxIdx(-1),
+      _comp(comp),
+      _root(new (_region) IDTNode(getNextGlobalIDTNodeIndex(), callTarget, symbol, -1, 1, NULL, budget)),
+      _indices(NULL),
+      _totalCost(0)
+   {   
+   increaseGlobalIDTNodeIndex();
+   }
+
+void TR::IDT::print()
+   {
+   bool verboseInlining = comp()->getOptions()->getVerboseOption(TR_VerboseInlining);
+   bool traceBIIDTGen = comp()->getOption(TR_TraceBIIDTGen);
+
+   if (!verboseInlining && !traceBIIDTGen)
+      return;
+   
+   // print header line
+   char header[1024];
+   const uint32_t candidates = getNumNodes() - 1;
+   sprintf(header,"#IDT: %d candidate methods inlinable into %s with a budget %d", 
+      candidates,
+      getRoot()->getName(comp()->trMemory()),
+      getRoot()->getBudget()
+   );
+
+   if (verboseInlining)
+      TR_VerboseLog::writeLineLocked(TR_Vlog_BI, header);
+   if (traceBIIDTGen)
+      traceMsg(comp(), "%s\n", header);
+
+   if (candidates <= 0) 
+      {
+      return;
+      }
+
+   // print the IDT nodes in BFS
+   TR::deque<TR::IDTNode*, TR::Region&> idtNodeQueue(comp()->trMemory()->currentStackRegion());
+
+   idtNodeQueue.push_back(getRoot());
+
+   while (!idtNodeQueue.empty())
+      {
+      TR::IDTNode* currentNode = idtNodeQueue.front();
+      idtNodeQueue.pop_front();
+
+      int32_t index = currentNode->getGlobalIndex();
+
+      // skip root node
+      if (index != -1) 
+         {
+         char line[1024];
+         sprintf(line, "#IDT: #%d: #%d inlinable @%d -> bcsz=%d %s target %s, static benefit = %d, benefit = %llu, cost = %d, budget = %d, callratio = %f, rootcallratio = %f", 
+            index,
+            currentNode->getParentGloablIndex(),
```

Comment:
looks like a typo here

Reply:
At first I wasn't sure how this typo made it through the check builds. It turns out that it's the same at the definition site